sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/component.py

```diff
@@ -449,23 +452,30 @@ def __call__(self, *args, **kwargs) -> [..., Union["Command", "Parallel"]]:
     @contextmanager
     def _resolve_local_code(self):
         """Resolve working directory path for the component."""
-        if hasattr(self, "code"):
-            code = getattr(self, "code")
-            # Hack: when code not specified, we generated a file which contains
-            # COMPONENT_PLACEHOLDER as code
-            # This hack was introduced because job does not allow running component without a
-            # code, and we need to make sure when component updated some field(eg: description),
-            # the code remains the same. Benefit of using a constant code for all components
-            # without code is this will generate same code for anonymous components which
-            # enables component reuse
-            if code is None:
-                with tempfile.TemporaryDirectory() as tmp_dir:
+        with tempfile.TemporaryDirectory() as tmp_dir:
```

Comment:
maybe `if hasattr(self, "code") and self.code is not None and os.path.isfile(code)` for quick return, then put other logics under a `with xx as tmp_dir`?
If cases are a little complicated now.

Reply:
refactor this for a quick yield path, shall looks better now.